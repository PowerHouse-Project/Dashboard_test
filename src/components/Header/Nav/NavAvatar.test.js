import React from "react";
import { render, screen } from "@testing-library/react";
import NavAvatar from "./NavAvatar";
import { profileImg } from "../../../assets/exports";

describe("NavAvatar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<NavAvatar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the correct profile image", () => {
    render(<NavAvatar />);
    const img = screen.getByAltText("Profile");
    expect(img).toHaveAttribute("src", profileImg);
  });

  it("displays the correct user name and role", () => {
    render(<NavAvatar />);
    expect(screen.getByText("Aditya")).toBeInTheDocument();
    expect(screen.getByText("Super User")).toBeInTheDocument();
  });

  it("has the correct class names", () => {
    render(<NavAvatar />);
    const listItems = screen.getAllByRole("listitem");
    expect(listItems[0]).toHaveClass("nav-item");
    expect(screen.getByRole("list")).toHaveClass("dropdown-menu");
  });
});
