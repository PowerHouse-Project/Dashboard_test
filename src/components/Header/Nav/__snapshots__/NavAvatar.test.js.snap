// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavAvatar renders correctly 1`] = `
<DocumentFragment>
  <li
    class="nav-item dropdown pe-3"
  >
    <a
      class="nav-link nav-profile d-flex items-center pe-0"
      data-bs-toggle="dropdown"
      href="#"
    >
      <img
        alt="Profile"
        class="rounded-full"
        src="profileImg.jpg"
      />
    </a>
    <ul
      class="dropdown-menu dropdown-menu-end dropdown-menu-arrow profile"
    >
      <li
        class="dropdown-header"
      >
        <h4
          class="font-jetBrains text-main-light-blue-dark"
        >
          Aditya
        </h4>
        <span>
          Super User
        </span>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <a
          class="dropdown-item d-flex items-center"
          href="users-profile.html"
        >
          <i
            class="bi bi-person"
          />
          <span
            class="pl-2"
          >
            My Profile
          </span>
        </a>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <a
          class="dropdown-item d-flex items-center"
          href="users-profile.html"
        >
          <i
            class="bi bi-gear"
          />
          <span
            class="pl-2"
          >
            Account Settings
          </span>
        </a>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <a
          class="dropdown-item d-flex items-center"
          href="users-profile.html"
        >
          <i
            class="bi bi-question-circle"
          />
          <span
            class="pl-2"
          >
            Tips and Tutorials
          </span>
        </a>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <a
          class="dropdown-item d-flex items-center"
          href="users-profile.html"
        >
          <i
            class="bi bi-box-arrow-right"
          />
          <span
            class="pl-2"
          >
            Log out
          </span>
        </a>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
    </ul>
  </li>
</DocumentFragment>
`;
